packages/styletron-engine-monolithic/src/server/server.js

```diff
@@ -0,0 +1,135 @@
+// @flow
+import type {StandardEngine} from "styletron-standard";
+import injectStylePrefixed from "../inject-style-prefixed.js";
+
+import type {
+  StyleObject,
+  FontFaceObject,
+  KeyframesObject,
+} from "styletron-standard";
+
+import {
+  keyframesToBlock,
+  keyframesBlockToRule,
+  fontFaceBlockToRule,
+  declarationsToBlock,
+  hashCssObject,
+} from "../css";
+
+export type sheetT = {|
+  css: string,
+  attrs: {[string]: string},
+|};
+
+export type optionsT = {
+  prefix?: string,
+};
+
+export type cacheT = {
+  [key: string]: string,
+};
+
+class StyletronServer implements StandardEngine {
+  styleCache: cacheT;
+  keyframesCache: cacheT;
+  fontFaceCache: cacheT;
+  opts: optionsT;
+
+  constructor(opts?: optionsT = {}) {
+    this.opts = opts || {};
+    this.styleCache = {};
+    this.fontFaceCache = {};
+    this.keyframesCache = {};
+  }
+
+  renderStyle(styles: StyleObject): string {
+    const className = hashCssObject(styles);
+    if (!this.styleCache[className]) {
+      this.styleCache[className] = injectStylePrefixed(
+        styles,
+        className,
+        this.opts.prefix || "",
+      ).join("");
+    }
+    return `${this.opts.prefix || ""}css-${className}`;
+  }
+
+  renderFontFace(fontFace: FontFaceObject): string {
+    const fontName = hashCssObject(fontFace);
+    if (!this.fontFaceCache[fontName]) {
+      this.fontFaceCache[fontName] = fontFaceBlockToRule(
+        `${this.opts.prefix || ""}font-${fontName}`,
+        declarationsToBlock(fontFace),
+      );
+    }
+    return `${this.opts.prefix || ""}font-${fontName}`;
+  }
+
+  renderKeyframes(keyframes: KeyframesObject): string {
+    const animationName = hashCssObject(keyframes);
+    if (!this.keyframesCache[animationName]) {
+      this.keyframesCache[animationName] = keyframesBlockToRule(
+        `${this.opts.prefix || ""}animation-${animationName}`,
```

Comment:
I used the same naming as emotion since I liked the readability. 

Brotli should take care of it - it even utilizes a dictionary with some common words where you can find `font` or `ation`. Also, keyframes and font-faces are relatively rare compare to other css rules.